node/core/pvf/common/src/worker/mod.rs

```diff
@@ -121,19 +121,18 @@ pub fn worker_event_loop<F, Fut>(
 				"Node and worker version mismatch, node needs restarting, forcing shutdown",
 			);
 			kill_parent_node_in_emergency();
-			let err: io::Result<Never> =
-				Err(io::Error::new(io::ErrorKind::Unsupported, "Version mismatch"));
-			gum::debug!(target: LOG_TARGET, %worker_pid, "quitting pvf worker({}): {:?}", debug_id, err);
+			let err = io::Error::new(io::ErrorKind::Unsupported, "Version mismatch");
+			worker_shutdown_message(debug_id, worker_pid, err);
 			return
 		}
 	}
 
 	// Delete all env vars to prevent malicious code from accessing them.
-	for (key, _) in std::env::vars() {
+	for (key, _) in std::env::vars_os() {
 		// TODO: *theoretically* the value (or mere presence) of `RUST_LOG` can be a source of
 		// randomness for malicious code. In the future we can remove it also and log in the host;
 		// see <https://github.com/paritytech/polkadot/issues/7117>.
-		if key != "RUST_LOG" {
+		if key.to_str() != Some("RUST_LOG") {
```

Comment:
Do we need `to_str` here? As far as I see, we can directly compare `OsStr` and `str`, even if `OsStr` variable has some unsafe unicode.

Reply:
Yep, looks like you're right.